shared/src/main/kotlin/com/egm/stellio/shared/model/ExpandedEntity.kt

```diff
@@ -109,20 +110,22 @@ class ExpandedEntity private constructor(
         }
     }
 
-    fun propertiesHaveAtMostOneDefaultInstance(): Boolean {
-        return properties.all { property ->
-            property.value.count { !it.containsKey(NGSILD_DATASET_ID_PROPERTY) } < 2
+    fun checkPropertiesHaveAtMostOneDefaultInstance() {
+        properties.forEach { property ->
+            if (property.value.count { !it.containsKey(NGSILD_DATASET_ID_PROPERTY) } > 1)
+                throw BadRequestDataException("Property ${property.key.extractShortTypeFromExpanded()} can't have more than one default instance")
         }
     }
 
-    fun propertiesHaveUniqueDatasetId(): Boolean {
-        return properties.all { property ->
+    fun checkPropertiesHaveUniqueDatasetId() {
+        properties.forEach { property ->
```

Comment:
``` property.find { ... } ?.let { throw ... } ```

Could this work ? (just a different style)

Reply:
property.first ?

Reply:
properties I meant

Reply:
I thought properties.first { } returned T? but actually it throws an exception if no element is found... what works is properties.find { } which actually returns a T?
so it should be :
```properties.find { ... } ?. let { throw ... }

see : https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/find.html & https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/first.html

Reply:
There isn't a `find` method on maps.

Current version may probably be slightly improved but yet it is simple and readable. So let's stick with it.